nanomq/rest_api.c

```diff
@@ -249,31 +253,51 @@ get_subscriptions(cJSON *data, http_msg *msg, uint64_t sequence)
 	http_msg res = { 0 };
 	res.status   = NNG_HTTP_STATUS_OK;
 
-	cJSON *res_obj;
-	cJSON *topics = cJSON_CreateArray();
-	res_obj       = cJSON_CreateObject();
+	cJSON *res_obj   = NULL;
+	cJSON *data_info = NULL;
+	data_info        = cJSON_CreateArray();
+	res_obj          = cJSON_CreateObject();
 	cJSON_AddNumberToObject(res_obj, "code", SUCCEED);
 	cJSON_AddNumberToObject(res_obj, "seq", (uint64_t) sequence);
 	cJSON_AddNumberToObject(res_obj, "rep", msg->request);
-	cJSON_AddItemToObject(res_obj, "subscriptions", topics);
-
-	size_t        sz = 0;
-	topic_queue **tq = dbhash_get_topic_queue_all(&sz);
+	cJSON_AddItemToObject(res_obj, "data", data_info);
+
+	dbtree *          db   = get_broker_db();
+	dbhash_ptpair_t **pt   = dbhash_get_ptpair_all();
+	size_t            size = cvector_size(pt);
+	for (size_t i = 0; i < size; i++) {
+		client_ctx *ctxt = (client_ctx *) dbtree_find_client(
+		    db, pt[i]->topic, pt[i]->pipe);
+
+		cJSON *data_info_elem;
+		data_info_elem = cJSON_CreateObject();
+		cJSON_AddItemToArray(data_info, data_info_elem);
+		const uint8_t *cid = conn_param_get_clientid(ctxt->cparam);
+		cJSON_AddStringToObject(data_info_elem, "Client ID", cid);
```

Comment:
Please remove the space between "Client" and "ID".

Reply:
Thanks for your opinion, it changed now.